### PR TITLE
Hand of God : A spin-off of cult [DON'T MERGE/93% Finished]

### PR DESCRIPTION
**Pull request by Neerti**
Main idea can be seen here. http://www.ss13.eu/phpbb/viewtopic.php?f=42&t=4184

This was polled and passed with flying colors on oldforums.

Thanks to Pete, Sawu, wb, RemieRichards, Iamgoofball for helping out with the code.
Thanks to Neinhaus, MrSnapwalk, Wjohnston for sprites.
Thanks to the people in #coderbus and #nanobus for answering my dumb code questions.

If I forgot someone let me know.

Remaining things to do:
- [ ] Objectives
  - [x] Follower Block objective
  - [ ] Deicide objective
  - [ ] Build objective
  - [x] Follower Escape objective
  - [ ] Sacrifice Prophet objective
- [ ] Bugfixes 
  - [ ] Fix god say code
  - [ ] Fix pylons shooting at gods
  - [ ] Fix admin player panel not showing deities
  - [ ] Fix structure building proc not stopping if a structure already exists
  - [ ] Fix rangefinding for abiltiies
  - [ ] Fix god icon
  - [ ] Fix follower counting
- [ ] Playtesting
  - [ ] Fix more bugs
- [ ] Merge conflict resolution

CURRENT BUGS: Carn yells at me from byond the code because I use transfer_to() a mob/camera, but the mind needs to transfer.  Blob somehow gets around this.  Not sure if it's worth it to make a snowflake transfer or to just remove the error as it doesn't appear to actually cause any damage.
The proc to build a structure doesn't stop if a structure already exists in it's location.  idk why atm.
Defensepylons try to shoot at gods, this is bad for obvious reasons.
Talking to other gods and prophets is broken.
Rangefinding for abilities doesn't work.
The icon for gods is visible to all humans.  This isn't intended.
Follower Counting isn't functioning.

<B>Remie Edit: Approved for use of Topic Procs</B>

### CODE REVIEW

**Comment by Atskadan** [comment]
neato

**Comment by GunHog** [comment]
Do you think /tg/ would accept this as well?

**Comment by Neerti** [comment]
Since my goal is to not touch cult assets as much as possible it might make it easier to port.

**Comment by PinkyGatchabee** [comment]
excited for this

**Comment by Neerti** [comment]
Red team is mostly done in terms of infrastructure.  Still need a more robust convert and deconvert proc.

**Comment by Neerti** [comment]
de/convert procs done.  Blue team needs love and buildings and stuff need to be completed.

I'd say I'm 65% done.

**Comment by Atskadan** [comment]
65% for 5 days of work is really fucking good.
Make sure you get other people to review it pretty thoroughly though; new gamemodes are a pretty big addition that absolutely needs to be bug-free.

**Comment by Neerti** [comment]
Pete fixed the problem with gods talking.

**Comment by Neerti** [comment]
Make sure to get Sawu and Goofball in the changelog credits for helping me a lot with bugs.

**Comment by Neerti** [comment]
Most structures and powers are finished.

**Comment by Neerti** [comment]
Defense pylons are almost finished, but currently suffer from attempting to shoot at a target they can't see, but is within range, as well as a few other minor bugs.

File: code/game/gamemodes/handofgod/powers.dm [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+		return
+
+
+/mob/camera/god/verb/build()
+	set category = "God Powers"
+	set name = "Create Structure (75)"
+	set desc = "Create the foundation of a divine object."
+
+	if(powerc(75,1))
+		var/choice = input("Choose what you wish to create.","Divine structure") as null|anything in list("ward","conduit","forge","convert alter",
```
Comment (GunHog) [inline code/game/gamemodes/handofgod/powers.dm]: altar

File: code/game/gamemodes/handofgod/structures.dm [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+	desc = "An altar dedicated to a deity.  Cultists can \"forcefully teach\" their non-aligned crewmembers to join their side and take up their deity."
+	icon_state = "convertaltar"
+
+	attack_hand(mob/living/user as mob)
+		var/mob/living/carbon/human/H = locate(/mob/living/carbon) in loc
+		if(!isfollower(user)) //Is the user a follower?
+			user << "<span class='notice'>You can't seem to do anything useful with this.</span>"
+			return
+		if(!H || !H.mind) //If nothing is on top of the alter, or the mob has no mind, stop the proc.
+			user << "<span class='danger'>Nobody sentient is on top of the alter, if anything at all.</span>"
```
Comment (GunHog) [inline code/game/gamemodes/handofgod/structures.dm]: altar

**Comment by Iamgoofball** [comment]
crusader greys II

**Comment by Atskadan** [comment]
crusader captains *

**Comment by Neerti** [comment]
Crusader Comdoms*

**Comment by Iamgoofball** [comment]
The Old Chaplains, Legacy of HoS

**Comment by Neerti** [comment]
Added about a third of the forge objects.

**Comment by vista-** [comment]
What's the current status on this?

**Comment by Neerti** [comment]
Making new forge objects (as in, not just childs of existing things) takes time, and I'd like to have a diverse range of things you can build.

Otherwise when forge is done I just got a few structures and powers to mop up, do roundstart antag assignment, objectives and I should be finished.

**Comment by GunHog** [comment]
Any chance of adding the mini-blob event from this: https://github.com/tgstation/-tg-station/pull/3582 to the list of possible events a high powered god can proc?  It still feels that the selection of events is too weak to be worthy of a god.

Also, is the God causing the event told what exactly it did?

**Comment by Neerti** [comment]
The god does not know which event it caused. Admins know, however.

**Comment by Neerti** [comment]
I'd say I'm 85 to 90% done now.

What remains: Objectives, a few major bugfixes involving defense pylons shooting at deities, range checks for some abilities, finishing abilities, sprites, and some tiny bug fixes.

File: code/game/gamemodes/handofgod/forgeobjs.dm [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+
+/obj/item/weapon/banner/grey/examine()
+	..()
+	if(user.mind.role == "Assistant")
+		usr << "GREYTIDE STATIONWIDE!!!!"
+	else if(user.mind.role == "Security Officer")
+		usr << "A horrible banner representing disorder and chaos."
+	else
+		usr << "A banner with a toolbox logo, likely assistant-related."
+*/
```
Comment (Atskadan) [inline code/game/gamemodes/handofgod/forgeobjs.dm:147]: why comment out ;_;

File: code/game/gamemodes/handofgod/forgeobjs.dm [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+/obj/item/clothing/suit/armor/vest/blessed
+	name = "blessed armor vest"
+	desc = "It look like all the other armor vests you've seen, but wearing it makes you feel slightly warm.  Maybe it's also better?"
+	armor = list(melee = 60, bullet = 25, laser = 60, energy = 20, bomb = 35, bio = 0, rad = 0)
+
+/obj/item/clothing/head/helmet/blessed
+	name = "blessed helmet"
+	desc = "A helmet that seems warm for some reason.  Not warm enough to protect you in space, but it's still nice.  Might be better too?"
+	armor = list(melee = 60, bullet = 25, laser = 60,energy = 20, bomb = 35, bio = 0, rad = 0)
```
Comment (Atskadan) [inline code/game/gamemodes/handofgod/forgeobjs.dm:156]: still an error here 

> 60,energy

Comment (RemieRichards) [inline code/game/gamemodes/handofgod/forgeobjs.dm:156]: Uh @Atskadan , that's not an error, byond will see the "," and immediately end what it was doing and move on, if it was an error, it would have been flagged up while Neerti compiled things.

Comment (Atskadan) [inline code/game/gamemodes/handofgod/forgeobjs.dm:156]: but no other armor values lack a space

File: code/game/gamemodes/handofgod/forgeobjs.dm (hunk 3 above)
Comment (Neerti) [inline code/game/gamemodes/handofgod/forgeobjs.dm:147]: It no compile.

And I cba how to make it check for role right now.

Comment (Atskadan) [inline code/game/gamemodes/handofgod/forgeobjs.dm:147]: if(istype(M, /mob/living/carbon/human) && M.job in list("Assistant")) ?
or if(user.mind && (user.mind.assigned_role == "Assistant"))

File: code/game/gamemodes/handofgod/forgeobjs.dm (hunk 4 above)
Comment (RemieRichards) [inline code/game/gamemodes/handofgod/forgeobjs.dm:156]: It's not an error, it compiles, there's no point arguing over a space that doesn't change how the code compiles.

Comment (Atskadan) [inline code/game/gamemodes/handofgod/forgeobjs.dm:156]: ill put a space in YOUR BRAIN :gun:

Comment (RemieRichards) [inline code/game/gamemodes/handofgod/forgeobjs.dm:156]: no

Comment (Atskadan) [inline code/game/gamemodes/handofgod/forgeobjs.dm:156]: try and stop me

**Comment by ZomgPonies** [comment]
Once this is ready for merge, please for the love of christ compress it down to a single commit.

**Comment by ZomgPonies** [comment]
As Atskadan removed his post saying "or, how about you don't barge into other people's codebases and try to tell them what to do to make it easier for you to steal it?" I'll answer on that since everyone got his rant by email anyhow.

First, get the stick out of your ass. If you don't want the code public, don't make the codebase public. Secondly from what I understand, this codebase was created due to dissent in the way /tg/ was being maintained. That means it was designed as much for downstream servers as for NTStation, hence the fact that 36 commits MIGHT be worth compressing, -especially- if it's modular and you intend to try to get it merged into /tg/.

Lastly as for me 'stealing' your public code, why don't you ask goofball who came into IRC not 6 hours ago to ask me if he could grab my new nations gamemode? Learn to play ball with the community.

tl;dr, go fuck yourself Atskadan.

**Comment by Atskadan** [comment]
I'm not going to get into an argument with you about how I disapprove of you or your codebase in this PR, and that's why I deleted my post. It's not appropriate to derail this PR over petty personal dislike, so I'm going to ask you to stop right now. If you want to argue, bring it somewhere else.

**Comment by Iamgoofball** [comment]
knock it off @dannno 

you're not helping at all

**Comment by Neerti** [comment]
Objectives are really confusing so this might take a bit.

**Comment by Neerti** [comment]
I'm still working on this, just fyi.

**Comment by Iamgoofball** [comment]
merge conflicts

i'm thinking a PR broke everyone's shit recently

**Comment by RemieRichards** [comment]
@Iamgoofball 

> [DON'T MERGE/90% Finished]

**Comment by Iamgoofball** [comment]
@RemieRichards

yeah but a fuckton of PRs suddenly got merge conflicts.

**Comment by RemieRichards** [comment]
@Iamgoofball No doubt my mutations overhaul, 79 files changed.

**Comment by Iamgoofball** [comment]
> 79 files changed

r.i.p

**Comment by Neerti** [comment]
I'm abandoning this project, the NTStation Project, and ss13, so this will never be finished.  Anyone's free to use the code to finish it or for whatever.
I'm sorry for disappointing everyone.

**Comment by vista-** [comment]
404 not found

**Comment by GunHog** [comment]
NEERTI WHY WHY WHY :-1:x :100:

**Comment by Iamgoofball** [comment]
RIP ARTYOM RIP ARTYOM CODE DELTA CODE DELTA

**Comment by optimumtact** [comment]
Happens to the best :tired_face:
